lang/LangPrimSource/SC_CoreMIDI.cpp

```diff
@@ -635,6 +637,9 @@ int prDisconnectMIDIIn(struct VMGlobals* g, int numArgsPushed) {
 
     return errNone;
 }
+
+UInt64 logicalTimeToNanos(struct VMGlobals* g) { return slotRawFloat(&g->thread->seconds) * 1000000000; }
```

Comment:
I would prefer the result of slotRawFloat to be pulled into a cast UInt64, the returning the result of that operation * your scaler. Also - since this scaler is hardcoded a number of times in the PR, I'd like to see it stored as a const somewhere to prevent changes in the future possibly messing up the scaler.